src/modals/confirm-sign-modal/use-sign-message.hook.ts

```diff
@@ -60,23 +61,30 @@ export const useSignMessage = (formValues: CreateNftFormValues) => {
 
   useEffect(() => {
     if (isString(signature)) {
-      from(fetchStableDiffusionSignIn({ timestamp, pk: account.publicKey, sig: signature }))
+      const subscription = from(getStableDiffusionAccessToken({ timestamp, pk: account.publicKey, sig: signature }))
         .pipe(
           switchMap(accessToken =>
-            from(fetchStableDiffusionGenerateArt({ accessToken, ...formValues })).pipe(
-              // TODO: Remove this imitation of Generate art
-              rxDelay(2000),
+            from(createStableDiffusionOrder(accessToken, formValues)).pipe(
               map(order => order),
-              tap(() => setIsLoading(false))
+              tap(() => {
+                dispatch(setAccessTokenAction({ accountPkh: account.publicKeyHash, accessToken }));
+                setIsLoading(false);
+              })
             )
-          )
+          ),
+          catchError(() => {
+            showWarningToast({ description: 'Ooops, something went wrong.\nPlease, try again later.' });
```

Comment:
API sometimes provides more informative errors messages (see https://github.com/madfish-solutions/stable-diffusion-nft-backend/blob/development/src/validators/order-creation.ts , https://github.com/madfish-solutions/stable-diffusion-nft-backend/blob/development/src/handlers/rate-limiters.ts#L45 , and https://github.com/madfish-solutions/stable-diffusion-nft-backend/blob/development/src/handlers/create-order.ts#L100). Please try using these messages